source/Node.Extensibility.Authentication/Extensions/ICanMatchExternalGroup.cs

```diff
@@ -0,0 +1,14 @@
+using System.Threading;
+using Octopus.Data.Model.User;
+
+namespace Octopus.Node.Extensibility.Authentication.Extensions
+{
+    public interface ICanMatchExternalGroup
+    {
+        /// <summary>
+        /// Returns an identity if one can be exactly matched to the given name
+        /// </summary>
+        /// <returns>Returns null if the provider is disabled, otherwise a result with a matching ExternalSecurityGroup if a single match was found. If none or more than 1 were found the group will be null.</returns>
+        ExternalSecurityGroupResult Match(string name, CancellationToken cancellationToken);
```

Comment:
`Match(sting name)` seems a bit too generic here, especially when [its implemented](https://github.com/OctopusDeploy/DirectoryServicesAuthenticationProvider/pull/9/files#diff-f2a352d8967b9e181c7592656caade51R135). The only way to know that its actually dealing with external groups is by looking at the return type.

Maybe 'ExternalGroupExists` or something like that?